SGoopas/Assets/Scripts/Game/2D/2D Player States/AirDashLeft2D.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace PlayerStates
+{
+    public class AirDashLeft2D : Base2DState
+    {
+        private float afterImageTime = 0.1f;
+        private float timeEllapsed;
+        private GameObject afterimage;
+        private Vector3 direction;
+        private IEnumerator coroutine;
+        private Rigidbody2D rigidbody;
+        private float origGrav;
+        public AirDashLeft2D(BasePlayerState previousState) : base(previousState)
+        {
+            direction = new Vector3();
+            PseudoConstructor();
+        }
+        public AirDashLeft2D(GameObject player, MasterPlayerStateMachine playerStateMachine, Transform groundCheck) : base(player, playerStateMachine, groundCheck)
+        {
+            direction = new Vector3();
+            PseudoConstructor();
+        }
+
+        public AirDashLeft2D(BasePlayerState previousState, Vector3 direction) : base(previousState)
+        {
+            this.direction = direction;
+            PseudoConstructor();
+        }
+        public AirDashLeft2D(GameObject player, MasterPlayerStateMachine playerStateMachine, Transform groundCheck, Vector3 direction) : base(player, playerStateMachine, groundCheck)
+        {
+            this.direction = direction;
+            PseudoConstructor();
+        }
+
+        private void PseudoConstructor()
+        {
+                afterimage = (GameObject)Resources.Load("AfterImage");
+                timeEllapsed = 0.0f;
+                rigidbody = PlayerObject.GetComponent<Rigidbody2D>();
+                origGrav = rigidbody.gravityScale;
+                rigidbody.gravityScale = 0.0f;
+                rigidbody.velocity = direction.normalized * DashDistance / DashTime;
+                coroutine = AfterImage();
+                if (MasterMonoBehaviour.Instance != null)
```

Comment:
this should really never be the case, but to be safe I think the best way to do this is to add a custom getter with an assertion that the instance isn't null

Reply:
I'm not sure I entirely understand what you're suggesting